master/internal/configpolicy/postgres_task_config_policy_intg_test.go

```diff
@@ -0,0 +1,509 @@
+	// Global scope.
+	w, _, ntscTCPs := CreateMockTaskConfigPolicies(ctx, t, pgDB, user, true, true, true)
+	ntscTCPs.Constraints = model.Constraints{}
+	expInvariantConfig, err := json.Marshal(ntscTCPs.InvariantConfig)
+	require.NoError(t, err)
+
+	count, err := db.Bun().NewSelect().
+		Table("task_config_policies").
+		Where("workspace_id IS NULL").
+		Count(ctx)
+	require.NoError(t, err)
+	require.Equal(t, 1, count)
+
+	_, err = db.Bun().NewInsert().Model(ntscTCPs).
+		Where("workspace_id = ?", nil).
+		Where("workspace_type = ?", model.ExperimentType).
+		Value("invariant_config", "?", string(expInvariantConfig)).
+		Exec(ctx)
+	require.ErrorContains(t, err, "duplicate key value violates unique constraint")
+
+	// Workspace-level.
+	w, _, ntscTCPs = CreateMockTaskConfigPolicies(ctx, t, pgDB, user, false, true, true)
+	ntscTCPs.Constraints = model.Constraints{}
+	expInvariantConfig, err = json.Marshal(ntscTCPs.InvariantConfig)
+	require.NoError(t, err)
+
+	count, err = db.Bun().NewSelect().
+		Table("task_config_policies").
+		Where("workspace_id = ?", w.ID).
+		Count(ctx)
+	require.NoError(t, err)
+	require.Equal(t, 1, count)
+
+	_, err = db.Bun().NewInsert().Model(ntscTCPs).
+		Where("workspace_id = ?", w.ID).
+		Where("workspace_type = ?", model.NTSCType).
+		Value("invariant_config", "?", string(expInvariantConfig)).
+		Exec(ctx)
+	require.ErrorContains(t, err, "duplicate key value violates unique constraint")
+}
+
+func TestDeleteConfigPolicies(t *testing.T) {
+	ctx := context.Background()
+	require.NoError(t, etc.SetRootPath(db.RootFromDB))
+	pgDB, cleanup := db.MustResolveNewPostgresDatabase(t)
+	defer cleanup()
+	db.MustMigrateTestPostgres(t, pgDB, db.MigrationsFromDB)
+
+	user := db.RequireMockUser(t, pgDB)
+	workspaceIDs := []int32{}
+
+	defer func() {
+		err := db.CleanupMockWorkspace(workspaceIDs)
+		if err != nil {
+			log.Errorf("error when cleaning up mock workspaces")
+		}
+	}()
+
+	tests := []struct {
+		name               string
+		global             bool
+		workloadType       model.WorkloadType
+		hasInvariantConfig bool
+		hasConstraints     bool
+		err                *string
+	}{
+
+		{"ntsc config no constraint", false, model.NTSCType, true, false, nil},
+		{"ntsc config and constraint", false, model.NTSCType, true, true, nil},
+		{"ntsc no config has constraint", false, model.NTSCType, false, true, nil},
+		{
+			"unspecified workload type", false, model.UnknownType, true, true,
+			ptrs.Ptr("invalid workload type"),
+		},
+		{"ntsc no config no constraint", false, model.NTSCType, false, false, nil},
+		{"global ntsc config no constraint", true, model.NTSCType, true, false, nil},
+		{"global ntsc config and constraint", true, model.NTSCType, true, true, nil},
+		{
+			"global unspecified workload type", true, model.UnknownType, true, true,
+			ptrs.Ptr("invalid workload type"),
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			switch test.workloadType {
+			case model.ExperimentType:
```

Comment:
The switch statement has experiment and default cases. Do we need one for NTSC (I think it's grouped with default as is)?

For experiment, I think we can remove the `case` for now, since it looks like none of the test cases use `model.ExperimentType`.

Reply:
Removed the case for experimentType in the delete test cases!